src/drt/src/pa/FlexPA_acc_point.cpp

```diff
@@ -1061,10 +1061,12 @@ void FlexPA::filterMultipleAPAccesses(
                     pin,
                     inst_term);
     if (is_std_cell_pin) {
-      has_access |= ((layer_num == router_cfg_->VIA_ACCESS_LAYERNUM
-                      && ap->hasAccess(frDirEnum::U))
-                     || (layer_num != router_cfg_->VIA_ACCESS_LAYERNUM
-                         && ap->hasAccess()));
+      const bool isViaAccessLayer
+          = layer_num == router_cfg_->VIA_ACCESS_LAYERNUM
+            || (layer_num >= router_cfg_->VIAINPIN_BOTTOMLAYERNUM
+                && layer_num <= router_cfg_->VIAINPIN_TOPLAYERNUM);
```

Comment:
This looks to repeat isViaAccessLayerNum and similarly in FlexPA::genPinAccessCostBounded.  Can this be a single shared function?